src/arch/riscv/regs/misc.hh

```diff
@@ -200,6 +201,42 @@ enum MiscRegIndex
     MISCREG_VTYPE,
     MISCREG_VLENB,
 
+    // H-extension registers (RV64)
+
+    MISCREG_HVIP, // vssi bit is alias for mip.vssi
+
+    MISCREG_MTINST,
+    MISCREG_MTVAL2,
+
+    MISCREG_HSTATUS,
+    MISCREG_HEDELEG,
+    MISCREG_HIDELEG,
+    MISCREG_HCOUNTEREN,
+    MISCREG_HGEIE,
+    MISCREG_HTVAL,
+    MISCREG_HTINST,
+    MISCREG_HGEIP,
+
+    MISCREG_HENVCFG,
+    MISCREG_HGATP,
+    MISCREG_HCONTEXT,
+    MISCREG_HTIMEDELTA,
+
+    MISCREG_VSSTATUS,
+    MISCREG_VSTVEC,
+    MISCREG_VSSCRATCH,
+    MISCREG_VSEPC,
+    MISCREG_VSCAUSE,
+    MISCREG_VSTVAL,
+    MISCREG_VSATP,
+
+    // This register stores the V-bit.
+    // It is not mandated by the specification,
+    // however it is useful for the implementation.
+    MISCREG_VIRT,
+
+    // H-extension registers end here
```

Comment:
You may need to update the checkpoint upgrade `util/cpt_upgraders`

Reply:
We have locally updated the cpt_upgrader but we have this failing test: test-gem5-library-riscv-hello-restore-checkpoint-ALL-x86_64-opt.

The code tries to restore some of the new MISCREGs for H-extension and crashes.

Do we have to upgrade the checkpoint beforehand? Would the code automatically try to do that?